packages/utils/src/commonwell/migrate-patients.ts

```diff
@@ -0,0 +1,149 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+// keep that ^ on top
+import { getEnvVarOrFail } from "@metriport/core/util/env-var";
+import { out } from "@metriport/core/util/log";
+import { sleep } from "@metriport/shared";
+import { APIMode, CommonWell, organizationQueryMeta } from "@metriport/commonwell-sdk";
+import axios from "axios";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { chunk } from "lodash";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { addOidPrefix } from "@metriport/core/domain/oid";
+
+/**
+ * This script performs a full patient migration in CommonWell:
+ * 1. Resets the patient's CW external data
+ * 2. Runs patient discovery
+ * 3. Deletes the patient in CommonWell using the old OID
+ *
+ * To run:
+ * 1. Set the env vars:
+ *  - CX_ID
+ *  - API_URL
+ *  - CW_OID (the old OID to use for deletion)
+ *  - CW_CERT (the CommonWell certificate)
+ *  - CW_KEY (the CommonWell private key)
+ *  - CW_ORG_NAME (the name of the organization)
+ * 2. Set the patientIds
+ * 3. Run the script with `ts-node src/commonwell/migrate-patients.ts`
+ */
+
+dayjs.extend(duration);
+
+const patientIds: string[] = [];
+const rerunPdOnNewDemographics = false;
+const cxId = getEnvVarOrFail("CX_ID");
+const apiUrl = getEnvVarOrFail("API_URL");
+
+const cwOid = getEnvVarOrFail("CW_OID");
+const cwCert = getEnvVarOrFail("CW_CERT");
+const cwKey = getEnvVarOrFail("CW_KEY");
+const apiMode = APIMode.integration;
+
+const PATIENT_CHUNK_SIZE = 2;
+
+const oldCwOid = "";
+const cwOrgName = "";
+
+const confirmationTime = dayjs.duration(10, "seconds");
+const delayTime = dayjs.duration(10, "seconds");
+
+type PdResponse = {
+  data: {
+    requestId: string;
+  };
+};
+
+function buildCWPatientId(orgOID: string, patientId: string): string {
+  return `${patientId}%5E%5E%5Eurn%3aoid%3a${orgOID}`;
+}
```

Comment:
This looks like a normal uri / url encoding. We can put whatever the real string is in there and then call encodeUriComponent on the whole string.

```
function buildCWPatientId(orgOID: string, patientId: string): string {
  return encodeUriComponent(
   `${patientId}^^^urn:oid:{orgOID}`;
  )
}
```

Reply:
Im going to leave as is to maintain the same pattern we've had in other places where we pass CW patient id.

Reply:
if pattern sgtm